File: app/src/main/java/shetj/me/base/fun/main/MainPresenter.java
package shetj.me.base.fun.main;

import me.shetj.base.base.BasePresenter;
import me.shetj.base.base.IView;

/**
 * <b>@packageName：</b> shetj.me.base.fun<br>
 * <b>@author：</b> shetj<br>
 * <b>@createTime：</b> 2018/10/29 0029<br>
 * <b>@company：</b><br>
 * <b>@email：</b> 375105540@qq.com<br>
 * <b>@describe</b><br>
 */
class MainPresenter extends BasePresenter<MainModel> {
	public MainPresenter(IView view) {
		super(view);
		model = new MainModel();
	}
}

File: app/src/main/java/shetj/me/base/fun/main/MainPresenter.kt
package shetj.me.base.`fun`.main

import androidx.appcompat.app.AppCompatDelegate
import me.shetj.base.base.BasePresenter
import me.shetj.base.base.IView

/**
 * **@packageName：** shetj.me.base.fun<br></br>
 * **@author：** shetj<br></br>
 * **@createTime：** 2018/10/29 0029<br></br>
 * **@company：**<br></br>
 * **@email：** 375105540@qq.com<br></br>
 * **@describe**<br></br>
 */
class MainPresenter(view: IView) : BasePresenter<MainModel>(view) {
    init {
        model = MainModel()
    }

    fun getNightModel(): Int {
        //android:forceDarkAllowed="true"
        //?android:attr/textColorPrimary 这是一种通用型文本颜色。它在浅色主题背景下接近于黑色，在深色主题背景下接近于白色。该颜色包含一个停用状态。
        //?attr/colorControlNormal 一种通用图标颜色。该颜色包含一个停用状态。
        //主题背景属性 ?attr/colorSurface 和 ?attr/colorOnSurface）轻松获取合适的颜色

        //浅色 - MODE_NIGHT_NO
        //深色 - MODE_NIGHT_YES
        //由省电模式设置 - MODE_NIGHT_AUTO_BATTERY
        //系统默认 - MODE_NIGHT_FOLLOW_SYSTEM
        val defaultNightMode = AppCompatDelegate.getDefaultNightMode()

        if (defaultNightMode == AppCompatDelegate.MODE_NIGHT_NO){
            return AppCompatDelegate.MODE_NIGHT_YES
        }
        if (defaultNightMode == AppCompatDelegate.MODE_NIGHT_YES){
            return AppCompatDelegate.MODE_NIGHT_NO
        }
        return AppCompatDelegate.MODE_NIGHT_YES
    }
}